### PR TITLE
Add `ThatAre[Not]Abstract`, `ThatAre[Not]Static` and `ThatAre[Not]Virtual` to `PropertyInfoSelector`

### PR DESCRIPTION
**Pull request by 94sedighi**
Part of #645

## IMPORTANT 

* [x] The code complies with the [Coding Guidelines for C#](https://www.csharpcodingguidelines.com/).
* [x] The changes are covered by unit tests which follow the Arrange-Act-Assert syntax and the naming conventions such as is used [in these tests](../tree/develop/Tests/FluentAssertions.Equivalency.Specs/MemberMatchingSpecs.cs#L51-L430).
* [x] If the PR adds a feature or fixes a bug, please update [the release notes](../tree/develop/docs/_pages/releases.md) with a functional description that explains what the change means to consumers of this library, which are published on the [website](https://fluentassertions.com/releases).
* [x] If the PR changes the public API the changes needs to be included by running [AcceptApiChanges.ps1](../tree/develop/AcceptApiChanges.ps1) or [AcceptApiChanges.sh](../tree/develop/AcceptApiChanges.sh).

### CODE REVIEW

**Review by jnyrup — COMMENTED** [review]
Small and simple, I like that :+1: 
Especially since it aligns the property selector with the method selector.

You'll need to add a line to the [release notes](https://github.com/fluentassertions/fluentassertions/blob/develop/docs/_pages/releases.md) with what has changed.
For inspiration, see 
https://github.com/fluentassertions/fluentassertions/pull/1740/files#diff-66bff3ed7b10da4ff2003461ce1c5a22828c018dfde1a22ef20d9c19a961936c

**Comment by 94sedighi** [comment]
Thanks for the rash review and sorry about the last PRs.
I have included many functionalities in one PR because you can review everything at once and thus you can save time. But as you also mentioned, it makes sense to make small tractable PRs.
As you see not all the tasks are finished for this PR, i will definitely extend the release notes and the docs.
And thanks for the link for inspiration :)

**Comment by jnyrup** [comment]
Now that the coveralls bot has posted the summary of code-coverage analysis, we can see that some code is not covered by tests.
> Overall coverage decreased (-0.009%) to 95.925%

If you click the top link its comment, and login to coveralls.io with your github account, it reports that `PropertyInfoExtensions.cs` has two branches that are not covered.

image
 
Click into  `PropertyInfoExtensions.cs` and you'll then see what's missing.
What's happening is that we're not using the coalescing part of the expression as the used test properties all have a getter.
 
image

**Review by dennisdoomen — APPROVED** [review]
Nothing additional to what @jnyrup already said

File: Src/FluentAssertions/Common/PropertyInfoExtensions.cs [hunk 1]
```diff
@@ -9,4 +9,16 @@ internal static bool IsVirtual(this PropertyInfo property)
         MethodInfo methodInfo = property.GetGetMethod(nonPublic: true) ?? property.GetSetMethod(nonPublic: true);
         return !methodInfo.IsNonVirtual();
     }
+
+    internal static bool IsStatic(this PropertyInfo property)
+    {
+        MethodInfo methodInfo = property.GetGetMethod(nonPublic: true);
```
Comment (jnyrup) [inline Src/FluentAssertions/Common/PropertyInfoExtensions.cs:15]: The removed code in 2007fa89db528c4cdfe2eb50d7cba2e0108c00a6 was correct, there just wasn't any unit tests using it.

It's quite nice, but it's valid to write a setter-only property
```cs
public class MyClass
{
    public static bool MyProperty { set; }
}
```

without ` ?? property.GetSetMethod(nonPublic: true)` `methodInfo` will now be `null` for a setter-only property and then invoking `IsStatic` on it throws a `NullReferenceException`.

File: docs/_pages/releases.md [hunk 2]
```diff
@@ -10,9 +10,13 @@ sidebar:
 ## Unreleased
 
 ### What's new
+* Added `ThatAre[Not]Abstract`, `ThatAre[Not]Static` and `ThatAre[Not]Virtual` properties for filtering in `PropertyInfoSelector.cs` - [#2054](https://github.com/fluentassertions/fluentassertions/pull/2054)
 * Added `BeOneOf` methods for object comparisons and `IComparable`s - [#2028](https://github.com/fluentassertions/fluentassertions/pull/2028)
 * Added `BeCloseTo` and `NotBeCloseTo` to `TimeOnly` - [#2030](https://github.com/fluentassertions/fluentassertions/pull/2030)
 
+### Fixes
+* Quering properties on classes, e.g. typeof(MyClass).Properties(), now also includes static properties - [#2054](https://github.com/fluentassertions/fluentassertions/pull/2054)
```
Comment (jnyrup) [inline docs/_pages/releases.md:18]:
```suggestion
* Quering properties on classes, e.g. `typeof(MyClass).Properties()`, now also includes static properties - [#2054](https://github.com/fluentassertions/fluentassertions/pull/2054)
```

**Review by jnyrup — COMMENTED** [review]
Just a note, see how I edited the PR title to exclude the "(as metnioned in More Assertions on types #645)".
image

The reference to #645 can be put in the PR description.
See e.g. https://cbea.ms/git-commit/ (just the first hit I found on Google) about the difference between commit summary and commit description.

We use a GitHub script to help generate [release notes](https://github.com/fluentassertions/fluentassertions/releases) from merged PRs, so by default the PR title will be shown on auto-generated release notes.

**Pull request merged**